hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1051,6 +1067,7 @@ private static <T> ClosableIterator<BufferedRecord<T>> getLogRecords(List<String
           .withMetaClient(datasetMetaClient)
           .withAllowInflightInstants(true)
           .withRecordBuffer(recordBuffer)
+          .withOptimizedLogBlocksScan(enableOptimizedLogBlocksScan)
```

Comment:
the flag is already set up in `getFileGroupReaderPropertiesFromStorageConf`

Reply:
Related to https://github.com/apache/hudi/pull/13591#discussion_r2233206936